src/workerd/io/worker-entrypoint.c++

```diff
@@ -257,20 +257,17 @@ kj::Promise<void> WorkerEntrypoint::request(
       // worker, not just for actors (and W2W below), but getting that right will require cleaning
       // up error handling more generally.
       return exceptionToPropagate();
-    } else KJ_IF_MAYBE(client, failOpenClient) {
+    } else KJ_IF_MAYBE(service, failOpenService) {
       // Fall back to origin.
 
       // We're catching the exception, but metrics should still indicate an exception.
       metrics->reportFailure(exception);
 
       auto promise = kj::evalNow([&] {
-        // kj::newHttpService adapts an HttpClient to look like an HttpService which makes it
-        // easier to forward the call.
-        auto httpWrapper = kj::newHttpService(**client);
-        auto promise = httpWrapper->request(
+        auto promise = service->get()->request(
             method, url, headers, requestBody, *wrappedResponse);
         metrics->setFailedOpen(true);
-        return promise.attach(kj::mv(httpWrapper), kj::mv(*client));
+        return promise.attach(kj::mv(*service));
       });
       return promise.catch_([this,wrappedResponse = kj::mv(wrappedResponse),
```

Comment:
Not specific to this PR, but does the fact that the lambda passed to `.catch_()` returns void mean that the promise is lazily evaluated?  I assume this doesn't affect behavior, though, since the returned promise is ultimately added to a TaskSet.

Reply:
Yes it would be lazy, but adding it to a `TaskSet` makes it eager.